components/post_view/acknowledgements/post_acknowledgements.tsx

```diff
@@ -0,0 +1,147 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import React, {memo, useState} from 'react';
+import {FormattedMessage} from 'react-intl';
+import classNames from 'classnames';
+import {useDispatch} from 'react-redux';
+import {
+    useFloating,
+    autoUpdate,
+    offset,
+    flip,
+    shift,
+    useHover,
+    useRole,
+    useInteractions,
+    safePolygon,
+    FloatingFocusManager,
+    useId,
+} from '@floating-ui/react-dom-interactions';
+
+import {CheckCircleOutlineIcon} from '@mattermost/compass-icons/components';
+
+import {acknowledgePost, unacknowledgePost} from 'mattermost-redux/actions/posts';
+
+import {Post, PostAcknowledgement} from '@mattermost/types/posts';
+import {UserProfile} from '@mattermost/types/users';
+
+import PostAcknowledgementsUserPopover from './post_acknowledgements_users_popover';
+
+import './post_acknowledgements.scss';
+
+type Props = {
+    acknowledgedAt: number;
+    currentUserId: UserProfile['id'];
+    hasReactions: boolean;
+    list?: Array<{user: UserProfile; acknowledgedAt: PostAcknowledgement['acknowledged_at']}>;
+    postId: Post['id'];
+}
+
+function moreThan5minAgo(time: number) {
+    const now = new Date().getTime();
+    return now - time > 5 * 60 * 1000;
+}
+
+function PostAcknowledgements({
+    acknowledgedAt,
+    currentUserId,
+    hasReactions,
+    list,
+    postId,
+}: Props) {
+    const dispatch = useDispatch();
+    const [open, setOpen] = useState(false);
+
+    const {x, y, reference, floating, strategy, context} = useFloating({
```

Comment:
I'm going to assume you know what you're doing here since I haven't looked at all into how these hooks are supposed to work 😛

Reply:
it's mostly by floating-ui documentation 😅